app/serializers/remote_search_form_serializer.rb

```diff
@@ -0,0 +1,27 @@
+class RemoteSearchFormSerializer < ActiveModel::Serializer
+  self.root = false
+
+  attributes :sort, :query
+
+  def sort
+    'registered_name'
+  end
+
+  def query
+    {
+      'filtered' => {
+        'filter' => {
+          'in' => {
+            'other_advice_methods' => remote_advice_method_ids
+          }
+        }
+      }
+    }
+  end
+
+  private
+
+  def remote_advice_method_ids
+    object.advice_methods.select(&:present?)
```

Comment:
Push this up into the form object instead. You can put this logic behind a meaningfully named predicate.